### PR TITLE
GSYE-463: Change gsy-e to run with python 3.11

### PR DESCRIPTION
**Pull request by BigTava**
## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=master
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -4,7 +4,7 @@ cached_property
 click
 click-default-group
 colorlog==4.7.2
-numpy==1.20.3
+numpy
```
Comment (hannesdiedrich) [inline requirements/base.in:7]: I remember issues with this. Do you mind finding out the version of numpy that was installed now on your env and pin it here? Thanks!

File: setup.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 gsy_framework_branch = "master"
 
 try:
-    with open("requirements/dev.txt", encoding="utf-8") as req:
+    with open("requirements/base.in", encoding="utf-8") as req:
```
Comment (hannesdiedrich) [inline setup.py:8]: Can you please explain the reason for this? 
1. We do not need the compiled versions of the requirements any more?
2. why going from dev to base requirements?

**Review by hannesdiedrich — COMMENTED** [review]
One comment and one question. Will follow up. Thanks!

File: requirements/base.in (hunk 1 above)
Comment (spyrostz) [reply requirements/base.in:7]: The issues were specifically with Pypy actually. If we use the normal Python interpreter, even the latest versions of numpy should work like a charm, so I think that this is ok!

File: setup.py (hunk 2 above)
Comment (spyrostz) [reply setup.py:8]: I also think that this is probably a remnant from testing. The dev requirements should be used when running `pip install -e .` (which is the command that this file is invoked) and also the compiled version is mandatory to be used in order to fix/hardcode the requirement versions.

File: src/gsy_e/models/area/scm_dataclasses.py [hunk 3]
```diff
@@ -295,7 +295,7 @@ def accumulate_fees_in_community(self, area_fees=AreaFees):
 @dataclass
 class AreaEnergyBills:  # pylint: disable=too-many-instance-attributes
     """Represents home energy bills."""
-    energy_rates: AreaEnergyRates = AreaEnergyRates(AreaFees())
+    energy_rates: AreaEnergyRates = field(default_factory=AreaEnergyRates(AreaFees))
```
Comment (spyrostz) [inline src/gsy_e/models/area/scm_dataclasses.py:298]: Thanks for catching this!

**Review by spyrostz — COMMENTED** [review]
Some comments, will LGTM once resolved

File: setup.py (hunk 2 above)
Comment (BigTava) [reply setup.py:8]: Thanks! Weird, I thought I removed this

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**